dot/utils_test.go

```diff
@@ -4,107 +4,518 @@
+	"github.com/ChainSafe/gossamer/lib/runtime"
+	"github.com/ChainSafe/gossamer/lib/runtime/wasmer"
+	"github.com/ChainSafe/gossamer/lib/utils"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-// TestNewConfig tests the NewTestConfig method
-func TestNewConfig(t *testing.T) {
+// newTestConfigWithFile returns a new test configuration and a temporary configuration file
+func newTestConfigWithFile(t *testing.T) (*Config, *os.File) {
 	cfg := NewTestConfig(t)
-	require.NotNil(t, cfg)
-}
 
-// TestNewConfigAndFile tests the NewTestConfigWithFile method
-func TestNewConfigAndFile(t *testing.T) {
-	testCfg, testCfgFile := newTestConfigWithFile(t)
-	require.NotNil(t, testCfg)
-	require.NotNil(t, testCfgFile)
+	configPath := filepath.Join(cfg.Global.BasePath, "config.toml")
+	err := os.WriteFile(configPath, nil, os.ModePerm)
+	require.NoError(t, err)
+
+	cfgFile := exportConfig(cfg, configPath)
+	return cfg, cfgFile
 }
 
-// TestInitNode
-func TestNewTestGenesis(t *testing.T) {
-	cfg := NewTestConfig(t)
-	require.NotNil(t, cfg)
+// newTestGenesisFile returns a human-readable test genesis file using "gssmr" human readable data
+func newTestGenesisFile(t *testing.T, cfg *Config) (filename string) {
+	fp := utils.GetGssmrGenesisPath()
+
+	gssmrGen, err := genesis.NewGenesisFromJSON(fp, 0)
+	require.Nil(t, err)
+
+	gen := &genesis.Genesis{
+		Name:       cfg.Global.Name,
+		ID:         cfg.Global.ID,
+		Bootnodes:  cfg.Network.Bootnodes,
+		ProtocolID: cfg.Network.ProtocolID,
+		Genesis:    gssmrGen.GenesisFields(),
+	}
+
+	b, err := json.Marshal(gen)
+	require.NoError(t, err)
 
-	genFile := NewTestGenesisRawFile(t, cfg)
+	filename = filepath.Join(t.TempDir(), "genesis.json")
+	err = os.WriteFile(filename, b, os.ModePerm)
+	require.NoError(t, err)
 
-	cfg.Init.Genesis = genFile
+	return filename
 }
 
-func TestNewTestGenesisFile(t *testing.T) {
-	cfg := NewTestConfig(t)
-	require.NotNil(t, cfg)
+func TestCreateJSONRawFile(t *testing.T) {
+	type args struct {
+		bs *BuildSpec
+		fp string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *os.File
+	}{
+		{
+			name: "working example",
+			args: args{
+				bs: &BuildSpec{genesis: NewTestGenesis(t)},
+				fp: "test_data/test.json",
+			},
+			want: &os.File{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := CreateJSONRawFile(tt.args.bs, tt.args.fp)
+			if tt.want != nil {
+				assert.NotNil(t, got)
+			}
+		})
+	}
+}
+
+func TestExportConfig(t *testing.T) {
+	type args struct {
+		cfg *Config
+		fp  string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *os.File
+	}{
+		{
+			name: "working example",
+			args: args{
+				cfg: &Config{},
+				fp:  "test_data/test.json",
+			},
+			want: &os.File{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := exportConfig(tt.args.cfg, tt.args.fp)
+			if tt.want != nil {
+				assert.NotNil(t, got)
+			}
```

Comment:
Assert

- the file path is as expected
- the file content is as expected

Same comment for below.

Reply:
Updated.